test/js/base/struct.js

```diff
@@ -34,4 +36,69 @@ describe("FileTuple", function() {
             assert.strictEqual(fileTuple instanceof Array, true);
         });
     });
+
+    describe("#fromString()", async function() {
+        beforeEach(function() {
+            if (typeof TextEncoder === "undefined") {
+                this.skip();
+            }
+        });
+
+        it("should be able to create a simple file tuple objects", () => {
+            const fileTuple = ripe.ripe.FileTuple.fromString("hello", "hello.txt", "text/plain");
+            assert.notStrictEqual(fileTuple, null);
+            assert.strictEqual(fileTuple.constructor, ripe.ripe.FileTuple);
+            assert.strictEqual(fileTuple.name(), "hello.txt");
+            assert.strictEqual(fileTuple.mime(), "text/plain");
+            assert.strictEqual(fileTuple.data().byteLength, 5);
+            assert.strictEqual(fileTuple instanceof ripe.ripe.FileTuple, true);
+            assert.strictEqual(fileTuple instanceof Array, true);
+        });
+    });
+
+    describe("#fromArrayBuffer()", async function() {
+        beforeEach(function() {
+            if (typeof TextEncoder === "undefined") {
+                this.skip();
+            }
+        });

```

Comment:
This one does not need the TextEncoder right?

Reply:
Nope, this needs to stay.

`fromArrayBuffer` calls `fromData`, which then call `encodeMultipart` which uses `TextEncoder` and it breaks if that doesn't exist.